### ISSUE TITLE
NullPointerException in KeySyncAdapterService

### ISSUE THREAD

**Issue by Valodim** [open-keychain/open-keychain#1573]
```
java.lang.RuntimeException: Unable to start service org.sufficientlysecure.keychain.service.KeyserverSyncAdapterService@21f63140 with null: java.lang.NullPointerException
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:2759)
	at android.app.ActivityThread.access$2100(ActivityThread.java:138)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1333)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:149)
	at android.app.ActivityThread.main(ActivityThread.java:5061)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:515)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:788)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:604)
	at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
	at org.sufficientlysecure.keychain.service.KeyserverSyncAdapterService.onStartCommand(KeyserverSyncAdapterService.java:80)
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:2742)
	... 10 more
```

the intent parameter seems to be null in `onStartCommand`. docs on the parameter:
> intent - The Intent supplied to Context.startService, as given. This may be null if the service is being restarted after its process has gone away, and it had previously returned anything except START_STICKY_COMPATIBILITY.

**Comment by adithyaphilip** [comment]
Looking into it

**Comment by dschuermann** [comment]
@adithyaphilip great :)

**Comment by adithyaphilip** [comment]
We return `START_NOT_STICKY` from `KeyserverSyncAdapterService`, and the docs for it say:
>if this service's process is killed while it is started (after returning from onStartCommand(Intent, int, int)), and there are no new start intents to deliver to it, then take the service out of the started state and don't recreate until a future explicit call to Context.startService(Intent). The service __will not receive a onStartCommand(Intent, int, int) call with a null Intent__ because it will not be re-started if there are no pending Intents to deliver.

So I'm not sure what exactly is causing it. Will keep investigating. Is there a way to reproduce it?

On a separate note, maybe we should be returning `START_REDELIVER_INTENT`(http://developer.android.com/reference/android/app/Service.html#START_REDELIVER_INTENT) instead, since if we died in a sync we wouldn't want to wait till the next periodic sync before retrying?

**Comment by dschuermann** [comment]
> So I'm not sure what exactly is causing it. Will keep investigating. Is there a way to reproduce it?
The stacktrace is from the Google developer console, so no, I don't know how to reproduce exactly :/

The only additional info are:
Android 4.4
MeMO Pad 8 

> On a separate note, maybe we should be returning START_REDELIVER_INTENT(http://developer.android.com/reference/android/app/Service.html#START_REDELIVER_INTENT) instead, since if we died in a sync we wouldn't want to wait till the next periodic sync before retrying?
Yup, sounds like a good idea.

**Comment by Valodim** [comment]
I think I was mistaken, it's likely not the intent which is null but `intent.getAction`. 

chcek out http://stackoverflow.com/questions/8421430/reasons-that-the-passed-intent-would-be-null-in-onstartcommand

**Comment by adithyaphilip** [comment]
Yup, I had found that link too. It does look like the action is null. I looked at all the parts in our code that start the service (with a find usage of `KeyserverSyncAdapterService.class`, this should get everything right?) and we seem to be setting the action everywhere. I'll keep looking to see if there's any way the action can be set but become null when passed to the service. Maybe it's due to an explicit intent from outside our code? I've verified that the sync adapter framework doesn't use an explicit intent to start the service (and therefore does not go to `onStartCommand`).

Probably not related, but is this by any chance from the same device that reported https://github.com/open-keychain/open-keychain/issues/1572?
